src/Haskell/Ide/Engine/Scheduler.hs

```diff
@@ -271,10 +259,10 @@ ideDispatcher
      -- ^ Callback to run for handling the request.
   -> Channel.OutChan (IdeRequest m)
      -- ^ Reading end of the channel where the requests are sent to this process.
-  -> IO void
-ideDispatcher stateVar caps env errorHandler callbackHandler pin =
+  -> IdeM void
+ideDispatcher env errorHandler callbackHandler pin =
   -- TODO: AZ run a single ReaderT, with a composite R.
```

Comment:
This TODO should be moved to the place where the code was moved to

Reply:
Should be fixed